cmd/mistryd/worker.go

```diff
@@ -116,9 +124,12 @@ func (s *Server) Work(ctx context.Context, j *Job) (buildInfo *types.BuildInfo,
 			}
 		}
 
-		// When there are no errors the LatestBuildPath can be updated
-		// to point to the last build. Otherwise it remains unchanged.
+		// if the build was successful, symlink it to the 'latest'
+		// path
 		if err == nil {
+			s.pq.Lock(j.Project)
```

Comment:
why lock?

Reply:
See https://github.com/skroutz/mistry/pull/107/commits/028a7d7fab1f023fd13b1d0d47fcd513921abc5c

Reply:
can you document the "why" in the code? e.g.  "atomically modify the latest link to avoid race condition by concurrent builds of the same project"